tensorflow/python/keras/metrics.py

```diff
@@ -2273,10 +2273,10 @@ def update_state(self, y_true, y_pred, sample_weight=None):
       Update op.
     """
     # Flatten the input if its rank > 1.
-    if y_pred.shape.ndims > 1:
+    if array_ops.rank(y_pred).numpy() > 1:
```

Comment:
Can we convert to tensor instead?

Reply:
yes, at first I thought of that only.

Reply:
Yes, a test case for the use case that wasn't supported before this change.